analyzer.go

```diff
@@ -1,50 +1,53 @@
 package lifecycle
 
 import (
-	"fmt"
-	"path/filepath"
-
 	"github.com/buildpacks/imgutil"
 	"github.com/pkg/errors"
 
+	"github.com/buildpacks/lifecycle/api"
 	"github.com/buildpacks/lifecycle/buildpack"
-	"github.com/buildpacks/lifecycle/launch"
+	"github.com/buildpacks/lifecycle/cmd"
 	"github.com/buildpacks/lifecycle/platform"
 )
 
 type Analyzer struct {
-	Buildpacks []buildpack.GroupBuildpack
-	LayersDir  string
-	Logger     Logger
-	SkipLayers bool
+	Image    imgutil.Image
+	Logger   Logger
+	Platform cmd.Platform
+
+	// Platform API < 0.7
+	Buildpacks    []buildpack.GroupBuildpack
+	Cache         Cache
+	LayerAnalyzer LayerAnalyzer
+	SkipLayers    bool
```

Comment:
I believe `SkipLayers` is only used by the `LayerAnalyzer` - maybe it could live there?

Reply:
(as a field on the struct)

Reply:
Done in 5facb022af2b458ffedf0b529226895072d629d2. Great suggestion!